vernac/declare.ml

```diff
@@ -2317,8 +2317,7 @@ let save_lemma_admitted_delayed ~pm ~proof =
     | UState.Monomorphic_entry _ -> false
     | UState.Polymorphic_entry _ -> true in
   let univs = UState.univ_entry ~poly uctx in
-  let sec_vars = if get_keep_admitted_vars () then proof_entry_secctx else None in
-  finish_admitted ~pm ~uctx ~pinfo ~sec_vars ~univs
+  finish_admitted ~pm ~uctx ~pinfo ~sec_vars:proof_entry_secctx ~univs
```

Comment:
What does this change have to do with this PR?
Shouldn't there be a test?
Is it consistent with the other use of get_keep_admitted_vars?

TBH I'm not sure if the behaviour of keep admitted vars is very sensible, is this really what we want?
~~~coq
Section S.
Variable x : nat.

Lemma foo : nat.
Admitted.

About foo.
(* uses x *)

Lemma bar : nat.
  clear x.
Admitted.

About bar.
(* uses nothing *)
~~~

Reply:
>  is this really what we want?

The current rule is to take the subset of section variables mentioned in the partial proof up to there and in the type. In `foo`, `x` occurs in the substitution of the main evar, that's why it is kept. I agree that this does not seem to do what we expect. We should skip instances of evars I guess.

> What does this change have to do with this PR?

It is indeed not strictly related to the "multiple" thing. It can go in another PR.